added mouse scroll wheel handling for win32 console

zewpo opened the pull request:
tested in both the old command prompt conhost.exe as well as the new windows terminal wt.exe

zewpo committed:
added mouse scroll wheel handling for win32 console

k-takata commented on src/os_win32.c:
```diff
@@ -1231,6 +1231,105 @@ mch_bevalterm_changed(void)
+	.  Otherwise, the wheel was rotated backward, toward the user.
+*/ 
+     static void
+decode_mouse_wheel( MOUSE_EVENT_RECORD *pmer )
```
```suggestion
    static void
decode_mouse_wheel(MOUSE_EVENT_RECORD *pmer)
```

k-takata commented on src/os_win32.c:
```diff
@@ -1231,6 +1231,105 @@ mch_bevalterm_changed(void)
+		clear_oparg(&oa);
+		cap.oap = &oa;
+		
+		if ( horizontal ) 
```
```suggestion
		if (horizontal)
```

k-takata commented on src/os_win32.c:
```diff
@@ -1231,6 +1231,105 @@ mch_bevalterm_changed(void)
+		modifiers |= MOD_MASK_ALT; // MOUSE_ALT;
+
+	// add (bitwise or) the scroll direction and the key modifier chars together.
+	g_nMouseClick = ( (direction << 8) | modifiers );
```
```suggestion
	g_nMouseClick = ((direction << 8) | modifiers);
```

k-takata commented on src/os_win32.c:
```diff
@@ -1303,9 +1402,15 @@ decode_mouse_event(
     if (g_nMouseClick != -1)
 	return TRUE;
+   
+    if (pmer->dwEventFlags == MOUSE_WHEELED || pmer->dwEventFlags == MOUSE_HWHEELED )
```
```suggestion
    if (pmer->dwEventFlags == MOUSE_WHEELED || pmer->dwEventFlags == MOUSE_HWHEELED)
```

k-takata commented on src/os_win32.c:
```diff
@@ -1939,12 +2044,37 @@ mch_inchar(
+		char_u modifiers = ((char_u *)(&g_nMouseClick))[0];
+		char_u scroll_dir = ((char_u *)(&g_nMouseClick))[1];
+
+		if( scroll_dir > 0 
```
```suggestion
		if (scroll_dir > 0
```

k-takata commented on src/os_win32.c:
```diff
@@ -1939,12 +2044,37 @@ mch_inchar(
+				scroll_dir==KE_MOUSEDOWN ||
+				scroll_dir==KE_MOUSEUP ||
+				scroll_dir==KE_MOUSELEFT ||
+				scroll_dir==KE_MOUSERIGHT ) 
```
```suggestion
				scroll_dir == KE_MOUSEDOWN ||
				scroll_dir == KE_MOUSEUP ||
				scroll_dir == KE_MOUSELEFT ||
				scroll_dir == KE_MOUSERIGHT)
```

k-takata commented on src/os_win32.c:
```diff
@@ -1939,12 +2044,37 @@ mch_inchar(
+				scroll_dir==KE_MOUSERIGHT ) 
+			)
+		{
+			if ( modifiers > 0 )
```
```suggestion
			if (modifiers > 0)
```

k-takata reviewed (commented):
Please also fix the indentation as already mentioned.

zewpo committed:
Update .gitignore

zewpo committed:
Update src/os_win32.c

Co-authored-by: K.Takata <kentkt@csc.jp>

zewpo committed:
Update src/os_win32.c

Co-authored-by: K.Takata <kentkt@csc.jp>

zewpo committed:
Update src/os_win32.c

Co-authored-by: K.Takata <kentkt@csc.jp>

zewpo committed:
Update src/os_win32.c

Co-authored-by: K.Takata <kentkt@csc.jp>

zewpo committed:
Update src/os_win32.c

Co-authored-by: K.Takata <kentkt@csc.jp>

zewpo committed:
Update src/os_win32.c

Co-authored-by: K.Takata <kentkt@csc.jp>

zewpo committed:
Update src/os_win32.c

Co-authored-by: K.Takata <kentkt@csc.jp>

zewpo committed:
Update .gitignore

Can we also gitignore /vim90/  because this will be created if following the INSTALLpc.txt

zewpo committed:
Update os_win32.c

fixed up a few indentation alignment issues caused by inconsistent use of tabs and spaces.  Actually, I'd slso like to fix up more in this file - in the areas that I did not touch for my main feature changes - so pending approval from maintainers - because that relates to previous code that I did not author.

zewpo committed:
Merge branch 'vim:master' into master

zewpo committed:
added a couple of gitignore entries under the windows section

zewpo committed:
Update os_win32.c

renamed l_col and l_row variables, because of VIm style guide suggest not to start with l_.

zewpo reviewed (commented):
ready for review again, I think.

zewpo committed:
Merge branch 'vim:master' into master

zewpo committed:
Update os_win32.c

removed redundant condition in if statement

zewpo committed:
Merge branch 'vim:master' into master
